### PR TITLE
Adjust RTCRtpSender API to match the specification

### PR DESCRIPTION
**Pull request by mmckenziedev**
https://www.w3.org/TR/webrtc/#dom-rtcrtpsender

### CODE REVIEW

File: types/webrtc/RTCPeerConnection.d.ts [hunk 1]
```diff
@@ -165,10 +165,10 @@ interface RTCRtcCapabilities {
...
 // https://www.w3.org/TR/webrtc/#dom-rtcrtpsender
 interface RTCRtpSender {
-    //readonly track?: MediaStreamTrack;
-    //readonly transport?: RTCDtlsTransport;
-    //readonly rtcpTransport?: RTCDtlsTransport;
-    setParameters(parameters?: RTCRtpParameters): Promise<void>;
+    readonly track?: MediaStreamTrack;
+    readonly transport?: RTCDtlsTransport;
+    readonly rtcpTransport?: RTCDtlsTransport;
+    setParameters(parameters: Partial<RTCRtpParameters>): Promise<void>;
```
Comment (dbrgn) [inline types/webrtc/RTCPeerConnection.d.ts]: Wow, I wasn't aware of partial types. Very nice.

**Review by dbrgn — COMMENTED** [review]
Looks good to me. I looked up why the fields were commented out (PR #12140):

> In the end I had to comment out some of the attributes because they clashed with types defined in `lib.es6.d.ts`. Not sure what the best approach would be here.

I assume that works by now?

**Comment by dbrgn** [comment]
Hm, @mmckenziedev, did you close this PR on purpose?

**Comment by mmckenziedev** [comment]
@dbrgn I closed it. There are some problems I need to resolve. Getting those conflict errors you mentioned. When I have the time I'll work on it and reopen this PR at some point.

**Pull request closed without merging**